tests/test_examples.py

```diff
@@ -0,0 +1,19 @@
+import os.path
+from openapi_spec_validator import validate_v2_spec_url
+from textwrap import dedent
+
+import yaml
+
+
+class TestExamples:
+    """
+        This test class should run the examples in the examples/ subdirectory, and then validate that the output
+        is as-expected, and that the generated schemas are valid.
+    """
+    def test_extended_example_with_marshmallow(self):
```

Comment:
I'd rather have this as a pytest test with the input coming from parametrization and fixtures.
If you need help, I can send you a PR with my suggestion.

Reply:
Could you describe this a bit more - what would you envisage passing in as the parameters? Happy to talk it through on here or take a PR - whichever is more convenient.